src/soundio/sounddeviceportaudio.cpp

```diff
@@ -430,27 +452,24 @@ SoundDeviceStatus SoundDevicePortAudio::close() {
             return SoundDeviceStatus::Error;
         }
 
-        // Stop the stream. Since the start of this function we are returning
-        // paAbort from the callback. paAbort stops the stream as soon as possible.
-        // When stopping the stream using this technique, we still need to call
-        // Pa_StopStream() before starting the stream again.
+        // Tell the process callback to return paAbort, which will cause
+        // PortAudio to stop the stream as soon as possible. This has the
+        // advantage over calling Pa_StopStream that it will not wait until all
+        // the buffers have been flushed, which can take a few (annoying)
+        // seconds when you're doing soundcard input.
+        m_callbackResult.store(paAbort, std::memory_order_release);
 
+        // We wait until the callback set with Pa_SetStreamFinishedCallback has
+        // been called. Without this, there is the risk of a deadlock when we
+        // call Pa_StopStream and Pa_CloseStream below.
+        waitForFinished();
+
+        // Call Pa_StopStream() in case the above failed.
         err = Pa_StopStream(pStream);
+
         // We should now be able to safely set m_pStream to null
```

Comment:
Before, DEBUG_ASSERT(Pa_IsStreamStopped())

Reply:
Yeah, I actually think calling Pa_StopStream is not needed anymore now, but I didn't want to remove it just in case.

Reply:
So we may do: 
```
VERIFY_OR_DEBUG_ASSERT(Pa_IsStreamStopped()) {
    err = Pa_StopStream(pStream);
}
``

Reply:
I have come to the conclusion that this is not the case. Pa_IsStreamStopped() is not true after the finished callback has been called, at least not on macOS, and also according to the documentation, after return paAbort from the callback, the finished callback will be called when the stream becomes _inactive_

Reply:
I can confirm.